### DISCUSSION TITLE
Nuxt re-deploy causes payload 404 not found

### DISCUSSION THREAD

**Discussion by gabrielsze** [nuxt/nuxt#9834] [Questions (Nuxt 2)]
I believe this was asked here a while back, so it's highly similar.

https://stackoverflow.com/questions/65577842/nuxt-js-app-redeploy-causes-payload-js-404-not-found

Essentially what happens is that on redeploy, there seem to be new payload.js files being loaded up, and those on the site which hasn't been updated cause a 404.

I have the same suspicions it has to do with some caching or the need to flush a new change to users on the "old static sites with old payload". Is there any way to work around such an issue?

**Comment by kissu** [comment]
Hi, it would probably be interesting where this one do happen, with which configuration, if it still happens while in a private window and if the build is okay locally. Or any relevant info at all actually.

**Reply by gabrielsze** [reply]
Hey @kissu, I spent some time looking into it deeper, and I suspect it could be due to some form of cache-control. You would notice in the screenshot that the payload seems to be cached for 4 hours - could this be an issue why there's a 404 error whenever there is a redeploy (new payloadjs)? This is on my production site so the error 404 pops up quite often (due to payload 404) and reloading it will definitely solve the issue.

I don't seem to be explicitly setting the cache control of payloadjs so i need to look into it. Could be firebase default though (?)
<img width="524" alt="Screenshot 2021-09-30 at 8 19 58 PM" src="https://user-images.githubusercontent.com/46477732/135453754-5d408a61-415e-4221-93c0-5ae553be3b55.png">
<img width="525" alt="Screenshot 2021-09-30 at 8 20 48 PM" src="https://user-images.githubusercontent.com/46477732/135453855-37346bbb-ede6-4566-95b3-d0c9846a679e.png">

**Comment by gabrielsze** [comment]
Seem to have found cloudflare caching my files for 4 hours, which matches the time. Should I be disabling caching for payload.js files?

<img width="992" alt="Screenshot 2021-09-30 at 9 19 44 PM" src="https://user-images.githubusercontent.com/46477732/135462720-b63395b4-f25d-4847-aad5-b4cf0bf46230.png">
